src/SecurityAdvisory/AdvisoryParser.php

```diff
@@ -0,0 +1,22 @@
+<?php declare(strict_types=1);
+
+namespace App\SecurityAdvisory;
+
+use Composer\Pcre\Preg;
+
+class AdvisoryParser
+{
+    public static function isValidCve(?string $cve): bool
+    {
+        return $cve && Preg::match('#^CVE-[0-9]{4}-[0-9]{4,}$#', $cve);
+    }
+
+    public static function titleWithoutCve(string $title): string
+    {
+        if (Preg::match('#^(CVE-[0-9x*?]{1,4}-?[0-9a-z-]+:)(.*)$#i', $title, $matches)) {
```

Comment:
The point of my `[0-9x*?]{1,4}-?[0x*?]+` changes was to support `*`, `?` and `0` as placheolder as well as `x`. I guess you need to add `*?` again to the second character class to make sure this works? Also probably can just simplify this a lot..

```suggestion
        if (Preg::match('#^(CVE-[0-9a-z*?-]+:)(.*)$#i', $title, $matches)) {
```

Reply:
oh, I see. Yeah, hard to predict what people will use in the future because they can't just reuse xxxx all year long.

Reply:
Yes, always assume the user is drunk :p